src/server/gql/schema/User.ts

```diff
@@ -0,0 +1,83 @@
+import { gql } from 'apollo-server-express'
+import { makeExecutableSchema } from 'graphql-tools'
+import { PrismaClient, User, Role } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+export default makeExecutableSchema({
+  typeDefs: gql`
+    type Query {
+      user(email: String!): User
+    }
+    type Mutation {
+      addUser(email: String!, givenName: String!, familyName: String!): User
+      assignRole(email: String!, role: String): User
+    }
+    type User {
+      id: Int
+      givenName: String!
+      familyName: String!
+      email: String!
+      roles: [String]
+    }
+  `,
+  resolvers: {
+    Query: {
+      user: async (_, { email }: User) =>
+        prisma.user.findUnique({
+          where: {
+            email,
+          },
+          include: {
+            userRoles: {
+              include: {
+                role: true,
+              },
+            },
+          },
+        }),
+    },
+    User: {
+      roles: (parent: { userRoles: { role: Role }[] }) => {
+        if (!parent) return []
+
+        const { userRoles = [] } = parent ?? {}
+        return userRoles.map(({ role }) => role.name)
+      },
+    },
+    Mutation: {
+      assignRole: async (
+        _,
+        { email, role }: { email: string; role: string }
+      ) => {
+        const [foundRole, foundUser] = await Promise.all([
+          prisma.role.findUnique({ where: { name: role } }),
```

Comment:
**Question**

Where is the `Role` entity defined?

Reply:
The `Role` entity is just a string. So that's why there's no definition, and why I'm just getting the name here.